include/flash_pm.h

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef ZEPHYR_INCLUDE_FLASH_PM_H_
+#define ZEPHYR_INCLUDE_FLASH_PM_H_
+
+
+#define FLASH_AREA DT_CHOSEN(zephyr_flash)
```

Comment:
Based on prefix these are related to the material that's defined in `<storage/flash_map.h>`.  Is there a reason they should not move there?

In any case the name `flash_pm.h` suggests it's related to power management for flash, which it isn't.

Reply:
I will move them.

Reply:
I spent some time trying to do this today and I believe there's a blocker here: this seems to lose the ability to work with multiple flash devices in the system, each with their own partitions.

Does the code need to be iterating over every node with compat `fixed_partitions` instead? Basically something like this:

```
/* in devicetree.h */
#define DT_FOREACH_NODE(compat, inst_expr) \
	COND_CODE_1(DT_HAS_COMPAT(compat), \
		    (UTIL_CAT(DT_FOREACH_INST_, compat)(inst_expr)), ())
// ... then redefine DT_INST_FOREACH(inst_expr) to be DT_FOREACH_NODE(DT_DRV_COMPAT, inst_expr)

/* flash_map.h */
#define GEN_AREAS_FOR_PARTITIONS_NODE(partition_inst) ...
enum flash_areas {
        DT_FOREACH_NODE(fixed_partitions, GEN_AREAS_FOR_PARTITIONS_NODE)
};
```

Where `GEN_AREAS_FOR_PARTITIONS_NODE(partition_inst)` works basically how `MACRO_MAP(FLASH_AREA_ID_GEN, FLASH_AREA_PARTITIONS_LIST)` is being used here, except it works on `DT_CHILDREN(DT_INST(partition_inst, fixed_partitions))` instead.